app/models/agents/twitter_stream_agent.rb

```diff
@@ -121,5 +122,117 @@ def remove_unused_keys!(agent, base)
         end
       end
     end
+
+    def self.setup_worker
+      if Agents::TwitterStreamAgent.dependencies_missing?
+        STDERR.puts Agents::TwitterStreamAgent.twitter_dependencies_missing
+        STDERR.flush
+        return false
+      end
+
+      Agents::TwitterStreamAgent.active.group_by { |agent| agent.twitter_oauth_token }.map do |oauth_token, agents|
+        filter_to_agent_map = agents.map { |agent| agent.options[:filters] }.flatten.uniq.compact.map(&:strip).inject({}) { |m, f| m[f] = []; m }
+
+        agents.each do |agent|
+          agent.options[:filters].flatten.uniq.compact.map(&:strip).each do |filter|
+            filter_to_agent_map[filter] << agent
+          end
+        end
+
+        config_hash = filter_to_agent_map.map { |k, v| [k, v.map(&:id)] } << oauth_token
+
+        Worker.new(id: agents.first.worker_id(config_hash),
+                   config: {filter_to_agent_map: filter_to_agent_map},
+                   agent: agents.first)
+      end
+    end
+
+    class Worker < LongRunnable::Worker
+      RELOAD_TIMEOUT = 10.minutes
+      DUPLICATE_DETECTION_LENGTH = 1000
+      SEPARATOR = /[^\w_\-]+/
+
+      def setup
+        require 'twitter/json_stream'
+        @filter_to_agent_map = @config[:filter_to_agent_map]
+      end
+
+      def run
+        @recent_tweets = []
+        EventMachine.run do
+          stream!(@filter_to_agent_map.keys, @agent) do |status|
+            handle_status(status)
+          end
+        end
+        Thread.stop
+      end
+
+      def stop
+        EventMachine.stop_event_loop if EventMachine.reactor_running?
+        thread.terminate
+      end
+
+      private
+      def stream!(filters, agent, &block)
```

Comment:
Not much, i just moved everything that [handles](https://github.com/cantino/huginn/blob/master/lib/twitter_stream.rb#L30-L34) a[status method to the [handle_status](https://github.com/cantino/huginn/pull/819/files#diff-b3a7232379661ce1bc6b7b557db2420dR210) method